File: Problems/665_Non-decreasing_Array.py
from typing import List

class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        count = 0
        pre_pre = pre = curr = float('-Inf')
        for num in nums:
            pre_pre = pre
            pre = curr
            curr = num
            if pre > curr:
                count += 1
                if count == 2:
                    return False
                if pre_pre>curr:
                    curr = pre
        return True
if __name__ == '__main__':
    sol = Solution()
    nums = [3,4,2,3]
    print(sol.checkPossibility(nums))

File: Problems/665_Non-decreasing_Array.cpp
#include<vector>
#include<math.h>

using namespace std;

class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int pre_pre = -pow(10, 5);
        int pre     = -pow(10, 5);
        int curr    = -pow(10, 5);
        int count   = 0;
        for(auto num:nums){
            pre_pre = pre;
            pre     = curr;
            curr    = num;
            if(pre>curr){
                count ++;
                if(count==2){
                    return false;
                }
                if(pre_pre>curr){
                    curr = pre;
                }
            }
        }
        return true;
    }
};